src/main/webapp/WEB-INF/view/activityfeed.jsp

```diff
@@ -28,84 +30,79 @@
 </head>
 <body>
 
-  <nav>
-    <a id="navTitle" href="/">CodeU Chat App</a>
-    <a href="/conversations">Conversations</a>
-    <a href="/about.jsp">About</a>
-  </nav>
+	<nav>
+		<a id="navTitle" href="/">CodeU Chat App</a> <a href="/conversations">Conversations</a>
+		<a href="/about.jsp">About</a>
+	</nav>
 
-  <div id="container">
-    <h1>Activity</h1>
+	<div id="container">
+		<h1>Activity</h1>
 
-  		<p>This is the activity feed page</p>
-  		
-  		<%-- This is User rendering --%>
-  		
-  		<%
-			List<User> users = (List<User>) request.getAttribute("users");
-			if (users == null || users.isEmpty()) {
-				return;	
-			} else {
-		%>
+		<p>This is the activity feed page</p>
 		<ul>
-			<%
-				for (User user : users) {
-			%>
-			<li><%=user.getCreationTime()%>: <%=user.getName()%> joined!</li>
-			<%
-				}
-			%>
-		
-		<%
-}
-%>
-
-  		<%-- This is Conversation rendering --%>
-  		
 		<%
-			List<Conversation> conversations = (List<Conversation>) request.getAttribute("conversations");
-			if (conversations == null || conversations.isEmpty()) {
+			HashMap<Instant,HashMap<UUID,String>> instantByInstance = (HashMap<Instant,HashMap<UUID,String>>) request.getAttribute("instantByInstance");
+			if (instantByInstance == null ) {
 				return;
 			} else {
-		%>
-		
-			<%
-				for (Conversation conversation : conversations) {
-			        String author = UserStore.getInstance().getUser(conversation.getOwnerId()).getName();
-			%>
-			<li><%=conversation.getCreationTime()%>: <%=author%> created a new conversation: <a href="/chat/<%= conversation.getTitle() %>"><%= conversation.getTitle() %></a></li>
+				String author;
+				Instant time;
+				String title;
+				for (Map.Entry<Instant,HashMap<UUID,String>> m:instantByInstance.entrySet()) {
+					HashMap<UUID,String> innerm = m.getValue();
+					System.out.println("printing..");
+					System.out.println("Instant:");
+					System.out.println(m.getKey().toString());
+					for (Map.Entry<UUID,String> im:innerm.entrySet()) {
+						System.out.println("UUID:");
+						System.out.println(im.getKey());
+						System.out.println("Section:");
+						System.out.println(im.getValue());
+						if ( im.getValue() == "user") { 
+							author = UserStore.getInstance().getUser(im.getKey()).getName();

```

Comment:
I think it would make more sense to declare a userStore variable, messageStore variable and conversationStore variable instead of calling .getInstance() each time.